internal/page/donor/onelogin_identity_details.go

```diff
@@ -0,0 +1,64 @@
+package donor
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/form"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type oneloginIdentityDetailsData struct {
+	App            page.AppData
+	Errors         validation.List
+	DonorProvided  *actor.DonorProvidedDetails
+	DetailsMatch   bool
+	DetailsUpdated bool
+	Form           *form.YesNoForm
+}
+
+func OneloginIdentityDetails(tmpl template.Template, donorStore DonorStore) Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request, donor *actor.DonorProvidedDetails) error {
+		data := &oneloginIdentityDetailsData{
+			App:            appData,
+			Form:           form.NewYesNoForm(form.YesNoUnknown),
+			DonorProvided:  donor,
+			DetailsUpdated: r.FormValue("detailsUpdated") == "1",
+			DetailsMatch: donor.Donor.FirstNames == donor.DonorIdentityUserData.FirstNames &&
+				donor.Donor.LastName == donor.DonorIdentityUserData.LastName &&
+				donor.Donor.DateOfBirth == donor.DonorIdentityUserData.DateOfBirth &&
+				donor.Donor.Address.Postcode == donor.DonorIdentityUserData.CurrentAddress.Postcode,
+		}
+
+		if r.Method == http.MethodPost {
+			if donor.DonorIdentityConfirmed() {
+				return page.Paths.ReadYourLpa.Redirect(w, r, appData, donor)
+			}
+
+			f := form.ReadYesNoForm(r, "yesIfWouldLikeToUpdateDetails")
+			data.Errors = f.Validate()
+
+			if data.Errors.None() {
+				if f.YesNo.IsYes() {
+					donor.Donor.FirstNames = donor.DonorIdentityUserData.FirstNames
+					donor.Donor.LastName = donor.DonorIdentityUserData.LastName
+					donor.Donor.DateOfBirth = donor.DonorIdentityUserData.DateOfBirth
+					donor.Donor.Address = donor.DonorIdentityUserData.CurrentAddress
```

Comment:
I think we should check the whole address above for a match, if we then go on to replace it all

Reply:
we could simplify/remove `identity.UserData` in favour of something simpler now we do this, and just track when the identity was confirmed on `DonorProvidedDetails`